tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,350 @@
+
+						for _, pod := range samplePodsCluster2.Items {
+							Eventually(common.GetObject).
+								WithArguments(ctx, clRemote, kube.Key(pod.Name, "sample"), &corev1.Pod{}).
+								Should(HaveCondition(corev1.PodReady, metav1.ConditionTrue), "Pod is not Ready on Cluster #2; unexpected Condition")
+						}
+						Success("Sample app is created in both clusters and Running")
+					})
+
+					It("can access the sample app from both clusters", func(ctx SpecContext) {
+						sleepPodNameCluster1, err := common.GetPodNameByLabel(ctx, clPrimary, "sample", "app", "sleep")
+						Expect(sleepPodNameCluster1).NotTo(BeEmpty(), "Sleep pod not found on Cluster #1")
+						Expect(err).NotTo(HaveOccurred(), "Error getting sleep pod name on Cluster #1")
+
+						sleepPodNameCluster2, err := common.GetPodNameByLabel(ctx, clRemote, "sample", "app", "sleep")
+						Expect(sleepPodNameCluster2).NotTo(BeEmpty(), "Sleep pod not found on  Cluster #2")
+						Expect(err).NotTo(HaveOccurred(), "Error getting sleep pod name on  Cluster #2")
+
+						// Run the curl command from the sleep pod in the  Cluster #2 and get response list to validate that we get responses from both clusters
+						Cluster2Responses := strings.Join(getListCurlResponses(sleepPodNameCluster2, kubeconfig2), "\n")
+						Expect(Cluster2Responses).To(ContainSubstring("Hello version: v1"), "Responses from  Cluster #2 are not the expected")
+						Expect(Cluster2Responses).To(ContainSubstring("Hello version: v2"), "Responses from  Cluster #2 are not the expected")
+
+						// Run the curl command from the sleep pod in the Cluster #1 and get response list to validate that we get responses from both clusters
+						Cluster1Responses := strings.Join(getListCurlResponses(sleepPodNameCluster1, kubeconfig), "\n")
+						Expect(Cluster1Responses).To(ContainSubstring("Hello version: v1"), "Responses from Cluster #1 are not the expected")
+						Expect(Cluster1Responses).To(ContainSubstring("Hello version: v2"), "Responses from Cluster #1 are not the expected")
+						Success("Sample app is accessible from both clusters")
+					})
+				})
+
+				When("sample apps are deleted in both clusters", func() {
+					BeforeAll(func(ctx SpecContext) {
+						// Delete the entire sample namespace in both clusters
+						Expect(kubectl.DeleteNamespace("sample", kubeconfig)).To(Succeed(), "Namespace failed to be deleted on Cluster #1")
+						Expect(kubectl.DeleteNamespace("sample", kubeconfig2)).To(Succeed(), "Namespace failed to be deleted on  Cluster #2")
+					})
+
+					It("sample app is deleted in both clusters and the namespace is empty", func(ctx SpecContext) {
+						common.CheckNamespaceEmpty(ctx, clPrimary, "sample")
+						common.CheckNamespaceEmpty(ctx, clRemote, "sample")
+						Success("Sample app is deleted in both clusters")
+					})
+				})
+
+				When("control plane namespace and gateway are deleted in both clusters", func() {
```

Comment:
I'm not sure what this is testing. Does the sail operator put a finalizer on the `Namespace`? Otherwise this is just testing that when you delete the namespace, everything inside of it gets deleted which we don't need to test. Maybe we need to test if you delete the `Istio` resource and _keep_ the `controlPlaneNamespace` that everything inside gets removed?

Reply:
I changed the name of the test to the test step and changed the idea of the step; the goal is to check that when the `Istio` resource is deleted, it deletes the listed pods, this is to verify that we do not have any issue deleting the related pod. I Move outside the deletion of the gateway and the namespace because this is more of a clean-up step to be able to iterate over the test again

Reply:
In fact just deleting the namespace as clean up should work in this cases. The only resource that we need to take care are the `Istio` and `RemoteIstio`